vyper/context/validation/module.py

```diff
@@ -58,34 +61,51 @@ def visit_AnnAssign(self, node):
         if name is None:
             raise VariableDeclarationException("Invalid module-level assignment", node)
 
-        elif name == "implements":
+        if name == "implements":
             interface_name = node.annotation.id
             self.namespace[interface_name].validate_implements(node)
-
-        else:
-            var = build_type_from_ann_assign(node.annotation, node.value)
-            if hasattr(var, "_member_of"):
-                try:
-                    self.namespace[var._member_of].add_member(name, var)
-                except VyperException as exc:
-                    raise exc.with_annotation(node) from None
-            elif node.get("annotation.func.id") == "constant":
-                # constants are added to the main namespace
-                try:
-                    self.namespace[name] = var
-                except VyperException as exc:
-                    raise exc.with_annotation(node) from None
-
-            else:
-                if node.value:
-                    raise VariableDeclarationException(
-                        "Storage variables cannot have an initial value", node.value
-                    )
-                # storage vars are added as members of self
-                try:
-                    self.namespace["self"].add_member(name, var)
-                except VyperException as exc:
-                    raise exc.with_annotation(node) from None
+            return
+
+        is_constant, is_public = False, False
+        annotation = node.annotation
+        if isinstance(annotation, vy_ast.Call):
+            # the annotation is a function call, e.g. `foo: constant(uint256)`
+            call_name = annotation.get("func.id")
+            if call_name in ("constant", "public"):
+                validate_call_args(annotation, 1)
+                if call_name == "constant":
+                    # declaring a constant
+                    is_constant = True
+
+                elif call_name == "public":
+                    # declaring a public variable
+                    is_public = True
+                # remove the outer call node, to handle cases such as `public(map(..))`
```

Comment:
Really want to do `Map[typ, typ]` syntax instead...

Reply:
Yeah - I didn't wanna add type checking and modify syntax at the same time though.  Already a big enough diff :smile: